File: 2526-find-consecutive-integers-from-a-data-stream/2526-find-consecutive-integers-from-a-data-stream.cpp
class DataStream {
public:
    int val = 0, kmax = 0;
    int valCount = 0;
    DataStream(int value, int k) {
        val = value;
        kmax = k;
    }
    
    bool consec(int num) {
        if(num == val)
            valCount++;
        else
            valCount = 0;
        return valCount >= kmax;
    }
};

File: 2526-find-consecutive-integers-from-a-data-stream/2526-find-consecutive-integers-from-a-data-stream.java
class DataStream {
    Queue<Integer> q=new LinkedList<>();
    int k;
    int value;
    public DataStream(int value, int k) {
        this.k=k;
        this.value=value;
    }
    
    public boolean consec(int num) {
        if(num!=value){q.clear();return false;}
        if(num==value){
            q.add(num);
        }
        if(q.size()==k){
            q.poll();
            return true;
        }
        return false;
    }
}